<launch>
    <arg name="robot" default="apollo.local"/>
    <arg name="circle" default="false" />

    <rosparam param="wheel_base">0.16</rosparam>
    <rosparam param="wheel_radius">0.033</rosparam>
    <rosparam param="left_wheel_joint">wheel_left_joint</rosparam>
    <rosparam param="right_wheel_joint">wheel_right_joint</rosparam>
    <rosparam param="odom_frame_id">odom</rosparam>
    <rosparam param="body_frame_id">base_footprint</rosparam>

    <include file="$(find nuturtle_description)/launch/load.launch" >
        <arg name="use_myOdom" value="true"/>
        <arg name="use_rviz" value="false"/>
    </include>

    <node name="teleop" pkg="turtlebot3_teleop" type="turtlebot3_teleop_key" unless="$(arg circle)"/>

    <node name="follow_circle" pkg="nuturtle_robot" type="nuturtle_robot_follow_circle" output="screen" if="$(arg circle)">
        <rosparam param="speed">0.1</rosparam>
        <rosparam param="radius">0.1</rosparam>
    </node>

    <machine name="turtlebot" user="ubuntu" address="$(arg robot)" env-loader="/home/ubuntu/loader.sh" default="true"/>

    <include file="$(find nuturtle_robot)/launch/basic_remote.launch" >
        <arg name="robot" value="$(arg robot)" />
    </include>

    <node machine="turtlebot" name="turtle_interface" pkg="nuturtle_robot" type="nuturtle_robot_turtle_interface" />
    <node machine="turtlebot" name="odometer" pkg="rigid2d" type="rigid2d_odometer" output="screen"/>

</launch>